src/main/java/java_echo_server/App.java

```diff
@@ -1,14 +1,57 @@
-/*
- * This Java source file was generated by the Gradle 'init' task.
- */
 package java_echo_server;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
 public class App {
-    public String getGreeting() {
-        return "Hello world.";
-    }
+    private static ServerSocket serverSocket;
+    private static Socket clientSocket;
 
     public static void main(String[] args) {
-        System.out.println(new App().getGreeting());
+        int portNumber = setPortNumber(args);
+        Client client = createClientSocketWrapper(portNumber);
+
+        EchoServer echoServer = new EchoServer(client);
+
+        do {
+            echoServer.start();
+        } while (echoServer.isRunning());
+        stopServer();
+    }
+
+    private static int setPortNumber(String[] terminal_args) {
+        return terminal_args.length > 0 ? Integer.parseInt(terminal_args[0]) : 1234;
+    }
+
+    private static Client createClientSocketWrapper(int port) {
+        try {
+            serverSocket = new ServerSocket(port);
+            clientSocket = serverSocket.accept();
+
+            Boolean autoFlushWriter = true;
+            PrintWriter printWriter = new PrintWriter(clientSocket.getOutputStream(), autoFlushWriter);
+            WriterWrapper outputStream = new PrintWriterWrapper(printWriter);
+            BufferedReader inputStream = new BufferedReader((new InputStreamReader(clientSocket.getInputStream())));
+
+            return new Client(inputStream, outputStream);
+
+        } catch (IOException e) {
+            System.err.println(e);
+            return null;
+        }
+    }
+
+    private static void stopServer() {
+        try {
+            System.out.println("The server is shutting down.");
+            clientSocket.close();
+            serverSocket.close();
+        } catch (IOException e) {
+            System.err.println("The server could not be shut down.");
+        }
     }
```

Comment:
This class is doing a lot of work. I'd consider splitting it up into two classes--one that's responsible for the main program entry point, and another that's responsible for the actual server behavior.